src/platform/packages/shared/kbn-unified-metrics-grid/src/components/chart/hooks/use_lens_props.test.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { renderHook, act, waitFor } from '@testing-library/react';
+import { BehaviorSubject } from 'rxjs';
+import { useLensProps } from './use_lens_props';
+import { useChartLayers } from './use_chart_layers';
+import type { LensSeriesLayer } from '@kbn/lens-embeddable-utils/config_builder';
+import { LensConfigBuilder } from '@kbn/lens-embeddable-utils/config_builder';
+import { dataViewPluginMocks } from '@kbn/data-views-plugin/public/mocks';
+import type { UnifiedHistogramServices } from '@kbn/unified-histogram';
+import type { UnifiedHistogramInputMessage } from '@kbn/unified-histogram/types';
+
+jest.mock('./use_chart_layers');
+jest.mock('@kbn/lens-embeddable-utils/config_builder');
+
+const LensConfigBuilderMock = LensConfigBuilder as jest.MockedClass<typeof LensConfigBuilder>;
+const useChartLayersMock = useChartLayers as jest.MockedFunction<typeof useChartLayers>;
+const servicesMock: Partial<UnifiedHistogramServices> = {
+  dataViews: dataViewPluginMocks.createStartContract(),
+};
+
+describe('useLensProps', () => {
+  const mockChartLayers: Array<LensSeriesLayer> = [
+    {
+      type: 'series',
+      seriesType: 'line',
+      xAxis: { field: '@timestamp', type: 'dateHistogram', colorPalette: 'foo' },
+      yAxis: [{ value: 'foo' }],
+    },
+  ];
+  const discoverFetch$ = new BehaviorSubject<UnifiedHistogramInputMessage>({ type: 'fetch' });
+  const timeRange = { from: 'now-1h', to: 'now' } as any;
```

Comment:
Nit: I think `as any` is not needed here

Reply:
yes! it could be replaced with `TimeRange`